### PR TITLE
Feature/state reducer

### PR DESCRIPTION
**Pull request by vasdee**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,4 @@ node_modules/
 
 # IDEs
 .vscode/
+.dir-locals.el
```
Comment (lilfolr) [inline .gitignore]: eww

File: src/demo.js [hunk 2]
```diff
@@ -2,11 +2,12 @@
 
 import React, {Component} from 'react';
 import moment from 'moment';
-
+import _ from 'lodash';
 import Timeline from './timeline';
 
 import {Layout, Form, InputNumber, Button, DatePicker} from 'antd';
 import 'antd/dist/antd.css';
+import TimelineItem from 'antd/lib/timeline/TimelineItem';
```
Comment (lilfolr) [inline src/demo.js]: Do we need this?

File: src/timeline.js [hunk 3]
```diff
@@ -163,104 +189,113 @@ export default class Timeline extends Component {
...
+            this.props.itemHeight * Math.round(pixToInt(selectedTarget.style['top']) / this.props.itemHeight)
+          );
+        });
+
+        this._grid.recomputeGridSize({rowIndex: 0});
       })
       .resizable({
         edges: {left: true, right: true, bottom: false, top: false}
       })
       .on('resizestart', e => {
```
Comment (lilfolr) [inline src/timeline.js]: Should we up the z-index on resize start?

File: src/demo.js (hunk 2 above)
Comment (vasdee) [reply src/demo.js]: I didn't add that... i swear. But git says otherwise :(

**Pull request merged**